src/foam/core/CSV.js

```diff
@@ -0,0 +1,233 @@
+          if ( o.outputCSV ) {
+            o.outputCSV(this)
+            return;
+          }
+          
+          // Outputs only the ordinal of ENUM
+          if ( foam.core.AbstractEnum.isInstance(o) ) {
+            this.outputPrimitive(o.ordinal);
+          } else {
+            var ps = o.cls_.getAxiomsByClass(foam.core.Property);
+
+            for ( var i = 0 ; i < ps.length ; i++ ) {
+              this.outputProperty(o, ps[i]);
+            }
+          }
+        },
+        Array: function(o) { /* Ignore arrays in CSV */ },
+        Function: function(o) { /* Ignore functions in CSV */ },
+        Object: function(o) { /* Ignore generic objects in CSV */ }
+      }, function(o) { this.outputPrimitive(o); })
+    },
+
+    function fromCSV(className, s) {
+      if ( ! s ) throw 'Invalid CSV input to convert. Arguments must be (className, csvString).'
+      var lines = s.split('\n');
+
+      if ( lines.length == 0 ) throw 'Insufficient CSV Input'
+
+      // Trims quotes and splits CSV row into array
+      var props = lines[0].split(',');
+
+      // Account for array of values (TODO?)
+      var values = lines[1].split(',');
+
+      // Calls for the creation of a new model
+      return this.createModel(props, values, className);
+    },
+
+    function createModel(props, values, className) {
+      foam.assert(props.length == values.length,
+        'Invalid CSV Input, header and value rows must be the same number of cells');
+
+      var model = foam.lookup(className).create();
+
+      for ( var i = 0 ; i < props.length ; ++i ) {
+        var p = props[i];
+        var v = values[i];
+
+        // Adds nested prop
+        if ( p.includes('__') ) {
+          p = p.split('__');
+          foam.assert(p.length >= 2, 'Invalid CSV object nesting, properties of inner objects are identified by `__`');
+          var prefix = p[0] + '__';
+
+          for ( var j = i ; j <= props.length ; ++j ) {
+            // If last element, or prefix no longer matches prop
+            if ( ( j == props.length ) || ( ! props[j].startsWith(prefix) ) ) {
+              // Creates a new model for the inner object
+              var prop = model.cls_.getAxiomByName(p[0]);
+              prop.set(model, createModel(props.slice(i, j).map(nestedProp => nestedProp.slice(prefix.length)), 
+                                          values.slice(i, j), prop.of.id));
+              
+              i = j - 1;
+              break;
+            }
+          }
+        // Adds regular prop
+        } else {
+          var prop = model.cls_.getAxiomByName(p);
+          prop.set(model, prop.of ? foam.lookup(prop.of.id).create({ ordinal: v }) : v);
+        }
+      }
+
+      return model;
+    }
+  ]
+});
+
+foam.LIB({
+  name: 'foam.csv',
+
+  constants: {
+    Standard: foam.csv.Outputter.create(),
+  },
+
+  methods: [
+    function toCSV(o) {
+      return foam.csv.Standard.toCSV(o);
+    },
+
+    function fromCSV(className, s) {
```

Comment:
I tried passing in the model (via `foam.lookup('fullQualifier')`), however, it gives the following error: `customModel.create is not a function`

When I pass in just the className, `foam.lookup('fullQualifier')` is exactly how I convert it into a model. I’m not sure what’s causing the error in that case.